Tasks/AzureFileCopy/AzureFileCopy.ps1

```diff
@@ -53,6 +55,67 @@ $resourceWinRMHttpPortKeyName = Get-ResourceHttpTagKey
 $resourceWinRMHttpsPortKeyName = Get-ResourceHttpsTagKey
 $skipCACheckKeyName = Get-SkipCACheckTagKey
 
+function Get-AzureStorageAccountResourceGroupName
+{
+    param([string]$storageAccountName)
+
+    Write-Verbose "(ARM)Getting resource details for azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType" -Verbose
+    $azureStorageAccountResourceDetails = Get-AzureResource -ResourceName $storageAccountName | Where-Object { $_.ResourceType -eq $ARMStorageAccountResourceType }
+    Write-Verbose "(ARM)Retrieved resource details successfully for azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType" -Verbose
+
+    $azureResourceGroupName = $azureStorageAccountResourceDetails.ResourceGroupName
+    if ([string]::IsNullOrEmpty($azureResourceGroupName) -eq $true)
+    {
+        Write-Verbose "Storage sccount: $storageAccountName not found" -Verbose
+        Throw (Get-LocalizedString -Key "Storage acccout: {0} not found. Please specify existing storage account" -ArgumentList $storageAccountName)
+    }
+
+    return $azureStorageAccountResourceDetails.ResourceGroupName
+}
+
+function Get-AzureStorageKeyFromARM
+{
+    param([string]$storageAccountName)
+
+    # get azure storage account resource group name
+    $azureResourceGroupName = Get-AzureStorageAccountResourceGroupName -storageAccountName $storageAccountName
+
+    Write-Verbose "(ARM)Retrieving storage key for the storage account: $storageAccount in resource group: $azureResourceGroupName" -Verbose
+    $storageKeyDetails = Get-AzureStorageAccountKey -ResourceGroupName $azureResourceGroupName -Name $storageAccount 
+    $storageKey = $storageKeyDetails.Key1
+    Write-Verbose "(ARM)Retrieved storage key successfully for the storage account: $storageAccount in resource group: $azureResourceGroupName" -Verbose
+
+    return $storageKey
+}
+
+function Get-AzureStorageKeyFromRDFE
+{
+    param([string]$storageAccountName)
+
+    Write-Verbose "(RDFE)Retrieving storage key for the storage account: $storageAccount" -Verbose
+    $storageKeyDetails = Get-AzureStorageKey -StorageAccountName $storageAccountName
+    $storageKey = $storageKeyDetails.Primary
+    Write-Verbose "(RDFE)Retrieved storage key successfully for the storage account: $storageAccount" -Verbose
+
+    return $storageKey
+}
+
+function Validate-AzurePowershellVersion
```

Comment:
I think even RG task has this common code, we should consider to move it to common plan in future.